diff --git a/air/tools/configurator/air/targets/sparc/leon4/config.py b/air/tools/configurator/air/targets/sparc/leon4/config.py
index 6efb7c3..0bd4439 100755
--- a/air/tools/configurator/air/targets/sparc/leon4/config.py
+++ b/air/tools/configurator/air/targets/sparc/leon4/config.py
@@ -128,8 +128,8 @@ arch_configure = air_sparc.get_sparc_configuration
 
 # IOP devices and definitions
 iop = IOP(defines=[],
-          devices=['greth0', 'greth1'],
-          drivers=['amba', 'greth'],
+          devices=['greth0', 'greth1', 'gr1553b0','grspw0'],
+          drivers=['amba', 'greth', 'gr1553', 'grspw', 'mil1553'],
           alias=dict(eth0='greth0',eth1='greth1'),
           arch=iop_arch)
 

